### ISSUE TITLE
Remove jnp.repeat for jax.pmap

### ISSUE THREAD

**Issue by AlexanderMath** [graphcore-research/pyscf-ipu#26]
https://github.com/graphcore-research/pyscf-ipu/blob/e42d60bd84931533a6cc04eb2aad57eb2b10342e/nanoDFT/experimental_pmap_nanoDFT.py#L250C18-L250C18

In this line we use jnp.repeat to add a (num_ipu, ..) axis to tensors we want repeated on all IPUs. 

TODO: 
- Refactor/remove jnp.repeat by using it the ["correct" jax way](https://jax.readthedocs.io/en/latest/_autosummary/jax.pmap.html) (? with in_axes?).
